firmware/shared/src/io/io_chimera.c

```diff
@@ -88,15 +95,28 @@ static AdcChannel io_chimera_parseNetLabelAdc(const AdcNetName *net_name)
     }
 }
 
-void io_chimera_init(UART *serial_uart, uint32_t name_gpio, uint32_t name_adc)
+void io_chimera_init(UART *serial_uart, uint32_t name_gpio, uint32_t name_adc, const Gpio *bootup_gpio)
 {
     uart             = serial_uart;
     is_mid_debug_msg = false;
     rx_packet_size   = 0;
     net_name_gpio    = name_gpio;
     net_name_adc     = name_adc;
 
-    hw_uart_receiveIt(uart, data, DEBUG_SIZE_MSG_BUF_SIZE);
+    n_chimera_gpio = bootup_gpio;
+
+    if (chimera_button_pressed())
+    {
+        hw_uart_receiveIt(uart, data, DEBUG_SIZE_MSG_BUF_SIZE);
+    }
+}
+
+void io_chimera_sleepTaskIfEnabled(void)
+{
+    if (chimera_button_pressed())
+    {
+        osDelay(osWaitForever);
+    }
```

Comment:
You should only check the button once, and then save the value as a static variable. I feel like there's a chance of confusing behavior happening if the button is pressed/released right as the board wakes up.

Reply:
Like its theoretically possible you press the button as a task starts, but not when it checks for the UART RX call.